onnxruntime/core/graph/contrib_ops/internal_nhwc_onnx_defs.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/constants.h"
+#include "core/graph/contrib_ops/contrib_defs.h"
+
+namespace onnxruntime {
+namespace internal_nhwc_onnx {
+
+void RegisterInternalNHWCOpset() {
+ONNX_CONTRIB_OPERATOR_SCHEMA(QLinearSoftmax)
```

Comment:
Some extra info. When the layout transformer runs (TransofrmLayoutForEP) it will replace the layout sensitive nodes with a node in the NHWC domain, and wrap Transpose operators around that node to convert from the original NCHW to NHWC. That limits the new nodes in the kMSInternalNHWCDomain domain to this list, along with a few ORT specific ones listed [here](https://github.com/microsoft/onnxruntime/blob/c40f73ae0cb986fd5e57c54764643adffbe2e37c/onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc#L804-L810).

Once that process finishes, the transpose optimizer will move the Transpose nodes and cancel as many out as possible, but that process does not involve changing the domain of the other nodes. Due to this the Softmax should stay in the ONNX domain.

Say we the EP asked for Conv -> Conv -> Softmax. First step would create Transpose(to NHWC) -> Conv(kMSInternalNHWCDomain) -> Transpose(to NCHW) -> Transpose(to NHWC) -> Conv(kMSInternalNHWCDomain) -> Transpose(to NCHW) -> Softmax.

Second step would cancel out most of the Transpose ops, and push the last one past the Softmax so it's at the boundary of the nodes requested by the EP to maximize the nodes the EP can run in NHWC layout. That results in Transpose(to NHWC) -> Conv(kMSInternalNHWCDomain) -> Conv(kMSInternalNHWCDomain) -> Softmax -> Transpose(to NCHW).

Reply:
Thanks for explaning.
On top of that, It should be fine to register only Layout-sensitive Ops in `kMSInternalNHWCDomain`, and OutputShape inferencing is the same as onnx Ops.